File: class2relational/src/main/java/org/atlanmod/model/dynamic/relationalModel/RelationalElement.java
package org.atlanmod.model.dynamic.relationalModel;

import org.atlanmod.model.DynamicElement;

public abstract class RelationalElement extends DynamicElement {

    abstract String getId();

    public RelationalElement(String classname) {
        super(classname);
    }

    @Override
    public boolean equals(Object obj) {
        if (obj instanceof RelationalElement){
            RelationalElement obj_ = (RelationalElement) obj;
            return this.getType().equals(obj_.getType()) && this.getId().equals(obj_.getId());
        }
        return false;
    }

}

File: sparktl-transformations/src/main/scala/org/atlanmod/class2relational/model/relationalmodel/RelationalElement.scala
package org.atlanmod.class2relational.model.relationalmodel

import org.atlanmod.tl.model.impl.dynamic.DynamicElement

import scala.collection.mutable

abstract class RelationalElement (classname: String) extends DynamicElement(classname, mutable.HashMap()) {

    def getId: String

    override def getType: String = classname


    override def weak_equals(o: Any): Boolean = {
        o match {
            case obj: RelationalElement => this.getType.equals(obj.getType)
            case _ => false
        }
    }

    override def equals(o: Any): Boolean = {
        o match {
            case obj: RelationalElement => this.getType.equals(obj.getType) && this.getId.equals(obj.getId)
            case _ => false
        }
    }
}